File: app/src/main/java/com/telolahy/towerofhanoi/manager/GameManager.java
package com.telolahy.towerofhanoi.manager;

import android.content.Context;
import android.content.SharedPreferences;

/**
 * Created by stephanohuguestelolahy on 11/8/14.
 */
public class GameManager {

    private static final GameManager INSTANCE = new GameManager();
    private static final String LEVEL_PREFS_KEY = "level";
    private static final String PREFS_NAME = "game_preferences";

    public static final int LEVELS_COUNT = 5;

    public static GameManager getInstance() {
        return INSTANCE;
    }

    public int currentLevel() {

        SharedPreferences prefs = preferences();
        int level = prefs.getInt(LEVEL_PREFS_KEY, 1);
        return level;
    }

    public void saveCurrentLevel(int level) {

        SharedPreferences prefs = preferences();
        SharedPreferences.Editor edit = prefs.edit();
        edit.putInt(LEVEL_PREFS_KEY, level);
        edit.commit();
    }

    private SharedPreferences preferences() {

        return ResourcesManager.getInstance().activity.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE);
    }
}

File: app/src/main/java/com/telolahy/towerofhanoi/manager/GameManager.kt
package com.telolahy.towerofhanoi.manager

import android.content.Context
import android.content.SharedPreferences
import com.telolahy.towerofhanoi.Constants.LEVELS_COUNT

/**
 * Created by stephanohuguestelolahy on 11/8/14.
 */
class GameManager {
  fun currentLevel(): Int {
    val prefs = preferences()
    return prefs.getInt(LEVEL_PREFS_KEY, 1)
  }

  fun saveCurrentLevel(level: Int) {
    val prefs = preferences()
    val edit = prefs.edit()
    edit.putInt(LEVEL_PREFS_KEY, level)
    edit.apply()
  }

  val isOnLastLevel: Boolean
    get() = currentLevel() == LEVELS_COUNT

  private fun preferences(): SharedPreferences {
    return ResourcesManager.instance.activity.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
  }

  companion object {
    val instance = GameManager()
    private const val LEVEL_PREFS_KEY = "level"
    private const val PREFS_NAME = "game_preferences"
  }
}